### PR TITLE
protocol.rs docs and unit tests

### PR DESCRIPTION
**Pull request by koraa**

### CODE REVIEW

**Comment by aparcar** [comment]
I'm a bit confused how `listen_loop` as somewhat retired via https://github.com/rosenpass/rosenpass/commit/b99d07287965728f03e9f942342dc6bd1feb51f9 but now it's somewhat re-added?

File: rosenpass/src/protocol/protocol.rs [hunk 1]
```diff
@@ -122,64 +192,281 @@ pub type PeerNo = usize;
...
+    /// The source of most timing information for the Rosenpass protocol
+    ///
+    /// We store most timing information in the form of f64 values, relative to a point stored in
+    /// this field.
     pub timebase: Timebase,
 
-    // Server Crypto
+    /// Static Secret Key Mine (our secret key)
     pub sskm: SSk,
+    /// Static Publi Key Mine (our public key)
```
Comment (aparcar) [inline rosenpass/src/protocol/protocol.rs:213]: Typo: Static Public Key

**Comment by koraa** [comment]
> I'm a bit confused how `listen_loop` as somewhat retired via [b99d072](https://github.com/rosenpass/rosenpass/commit/b99d07287965728f03e9f942342dc6bd1feb51f9) but now it's somewhat re-added?

Retiring ListenLoop was a regression.

File: rosenpass/src/protocol/protocol.rs [hunk 2]
```diff
@@ -27,7 +32,7 @@ use rosenpass_to::To;
 use rosenpass_util::functional::ApplyExt;
 use rosenpass_util::mem::DiscardResultExt;
 use rosenpass_util::{cat, mem::cpy_min, time::Timebase};
-use zerocopy::{AsBytes, FromBytes, Ref};
+use zerocopy::{AsBytes, FromBytes, FromZeroes, Ref};
```
[the remark this answers is not in the document]
Comment (koraa) [reply rosenpass/src/protocol/protocol.rs:35]: Can you push a fix to my branch directly?

File: rosenpass/src/protocol/protocol.rs (hunk 1 above)
Comment (koraa) [reply rosenpass/src/protocol/protocol.rs:213]: Can you push a fix to my branch directly?

File: rosenpass/src/protocol/protocol.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (aparcar) [reply rosenpass/src/protocol/protocol.rs]: Done.

File: rosenpass/src/protocol/protocol.rs (hunk 1 above)
Comment (aparcar) [reply rosenpass/src/protocol/protocol.rs]: Done.

**Comment by koraa** [comment]
@aparcar merged.

**Pull request merged**